lib/cequel/schema/table_reader.rb

```diff
@@ -150,18 +173,21 @@ def read_table_settings
       end
 
       def table_data
-        @table_data ||=
-          begin
-            cluster = keyspace.cluster
-            cluster.refresh_schema
+        @table_data ||= cluster && cluster
+          .keyspace(keyspace.name)
+          .table(table_name.to_s)
+      end
 
-            fail(NoSuchKeyspaceError, "No such keyspace #{keyspace.name}") if
-              !cluster.has_keyspace?(keyspace.name)
+      def cluster
+        @cluster ||= begin
+          @cluster = keyspace.cluster
+          @cluster.refresh_schema
```

Comment:
I don't think this is a show stopper. It does only happen the first time, due to the memoization.